bench/run_bench.py

```
@@ -0,0 +1,576 @@
+    """Run `cmd`, returning (instr, cycles, elapsed, pmc_events) for `pid`.
+
+    instr/cycles are None when no counter backend is available. `pmc_events` is
+    the extra branch/L1D dict, which only the macOS `pmc_tool` path fills.
+    """
+    if COUNTER_BACKEND == "rusage":
+        i0, c0 = rusage(pid)
+        if pmc_ok:
+            ev, dt = pmc_run(pmc, cmd)
+        else:
+            t0 = time.time()
+            subprocess.run(cmd, capture_output=True)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)